### PR TITLE
Initial home desc setup with added seeding to DB

### PR DESCRIPTION
**Pull request by Supra-MS**
@amandawinkles @peter1000130,

 I have,
- added the seeding script for inserting 100 sample datas to the DB
- created endpoints as below and updated in the doc as same

> The below endpoint serves data to other dependent modules like Photos, Check-In
> /rooms/:roomId/title 
> /rooms/:roomId/overview

> The below endpoint serves data to self
> /rooms/:roomId/
> /rooms/:roomId/amenities
> /rooms/:roomId/description
> /rooms/:roomId/highlights
> /rooms/:roomId/sleeping-arranges

- created the below json accessed from s3, for our ease of development independently

> https://fecgai-homedesc.s3-us-west-1.amazonaws.com/json/title.json
> https://fecgai-homedesc.s3-us-west-1.amazonaws.com/json/overview.json

> https://fecgai-homedesc.s3-us-west-1.amazonaws.com/json/amenities.json
> https://fecgai-homedesc.s3-us-west-1.amazonaws.com/json/description.json
> https://fecgai-homedesc.s3-us-west-1.amazonaws.com/json/highlights.json
> https://fecgai-homedesc.s3-us-west-1.amazonaws.com/json/sleeping-arranges.json

> https://fecgai-homedesc.s3-us-west-1.amazonaws.com/json/rooms.json
> https://fecgai-homedesc.s3-us-west-1.amazonaws.com/json/hostInfo.json

- able to get the data from client side (working API)
- amenities, for now sending as object to play around/handle from client side (tbd)

Please review.
Updating Trello ticket: https://trello.com/c/Wcgu2Y7w/6-link-to-pr

### CODE REVIEW

File: HostHomeDescription/client/src/components/HomeDescription.js [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+  }
+
+  componentDidMount() {
+    this.getHomeDescById(3);
+  }
+
+  getHomeDescById(id) {
+    var serverUrl = 'http://localhost:3002'
+    http.get(`${serverUrl}/rooms/${id}`)
+      .then(response => {
```
Comment (peter1000130) [inline HostHomeDescription/client/src/components/HomeDescription.js:21]: @Supra-MS  -- just as a quick thought --> have you considered using the async await pattern for promises? I've found it to be super useful for shortening code length.

Comment (peter1000130) [reply HostHomeDescription/client/src/components/HomeDescription.js:21]: I've never used axios, so not sure if it supports that

File: HostHomeDescription/server/helpers/choosePlaceTypeWithLogic.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+var selectedPlaceType = choosePlaceType();
+
+var placeTypeLogic = (placeType) => {
+  let is_entire_home, just_bath, is_private_bath, is_shared_bath;
+  if (placeType === 'Entire place') {
+    is_entire_home = true;
+    just_bath = true;
+    is_private_bath = false;
+    is_shared_bath = false;
+    return { placeType, is_entire_home, just_bath, is_private_bath, is_shared_bath }
```
Comment (peter1000130) [inline HostHomeDescription/server/helpers/choosePlaceTypeWithLogic.js:19]: I appreciate that you used object shorthand here! Makes the code clean

File: HostHomeDescription/server/helpers/cleanLogic.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  var getRandomCancellationDays = () => {
+    var index = Math.floor(Math.random() * freeCancellationDays.length);
+    return freeCancellationDays[index];
+  };
+
+  var recentCleanReviews = faker.random.number({min: 0, max: 20});
+
+  var cleanLogicObj = {
+    freeCancellationDays: getRandomCancellationDays(),
+    recentCleanReviews: 0,
```
Comment (peter1000130) [inline HostHomeDescription/server/helpers/cleanLogic.js:15]: Sanity checking: was setting the preset clean reviews = 0 a deliberate choice? Wondering why not set this value to the previously declared variable?

File: HostHomeDescription/server/index.js [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+    .catch(err => {
+      console.log('Error retrieving home description id - Title: ', err);
+      res.status(500).send({ message: 'Error retrieving home description id - Title' });
+    });
+});
+
+app.get('/rooms/:roomId/overview', (req, res) => {
+  console.log('Parameter send by id in the req: ', req.params);
+  HomeDescription.findById(req.params.roomId)
+    .then(data => {
```
Comment (peter1000130) [inline HostHomeDescription/server/index.js:65]: Async await would really help flatten these promises!

File: HostHomeDescription/server/index.js [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+      }
+    })
+    .catch(err => {
+      console.log('Error retrieving home description id - amenities: ', err);
+      res.status(500).send({ message: 'Error retrieving home description id - amenities' })
+    });
+});
+
+app.listen(PORT, () => {
+  console.log(`Server listening on port http://localhost:${PORT}`);
```
Comment (peter1000130) [inline HostHomeDescription/server/index.js:148]: Oh interesting! I've never placed app.listen at the end of the index.js file before. Does the placement of either beginning / the end make any difference?

File: HostHomeDescription/client/src/components/HomeDescription.js (hunk 1 above)
Comment (Supra-MS) [reply HostHomeDescription/client/src/components/HomeDescription.js:21]: Oh yeah. I'm using that in writing test cases for now. Thanks for bringing that up 👍  .. I need more practice on that :)

**Review by peter1000130 — APPROVED** [review]
Overall looks good! Very impressed by the fact that you've generated so much code in such a short period of time. I have no clue how you were able to piece together SO MANY different data fields in mongo. There was... so... much data. One small thing that might be helpful is to try out the Async Await pattern :)

File: HostHomeDescription/client/src/components/HomeDescription.js (hunk 1 above)
Comment (Supra-MS) [reply HostHomeDescription/client/src/components/HomeDescription.js:21]: axios is a promised based lib. 
From the axios doc: https://www.npmjs.com/package/axios
> async function getUser() {
  try {
    const response = await axios.get('/user?ID=12345');
    console.log(response);
  } catch (error) {
    console.error(error);
  }
}

https://stackabuse.com/making-asynchronous-http-requests-in-javascript-with-axios/#:~:text=Axios%20is%20a%20Promised%2Dbased,design%20patterns%20in%20this%20tutorial.

File: HostHomeDescription/server/helpers/cleanLogic.js (hunk 3 above)
Comment (Supra-MS) [reply HostHomeDescription/server/helpers/cleanLogic.js:15]: If we have recentReviews > 10 then isClean is true & vice versa.
So I am setting that to 0 initially then returning after the update.

File: HostHomeDescription/server/index.js (hunk 5 above)
Comment (Supra-MS) [reply HostHomeDescription/server/index.js:148]: It doesn't make any difference I guess, not sure though [ as createServer itself is async ].  
Except for better coding practices like we say import the needed files, packages, connect to DB, router etc only then start listening [we write code async :) , I suppose ]

**Pull request merged**